projects/angular-cld/src/lib/cloudinary-image.component.ts

```diff
@@ -174,6 +182,6 @@ export class CloudinaryImage
   }
 
   accessibilityModeHandler() {
-    return this.cloudinary.url(this.publicId, {transformation: [this.options, accessibilityEffect[this.accessibility]]});
+    return this.cloudinary.url(this.publicId, {transformation: [this.options, accessibilityEffect[this.accessibility]], accessibility: true, ...analyticsOptionsDefault});
```

Comment:
should we always use analyticsOptionsDefault here?

can't this get extended?

Reply:
Why do we need to send the default values all around?
We can just use them in a single place and allow the user sending just overriding values. 
Meaning, you can do: 
`
return this.cloudinary.url(this.publicId, {transformation: [this.options, accessibilityEffect[this.accessibility]]}, {analytics.features.accessibility: true});
`

Reply:
I think that's the confusing part, these are not default values, they are the SDK values and cannot be overwritten by the user.

Reply:
Yeah I think the confusion was that you ccan overwrite these values as they are defaults. I'll change the name to make it clearer. These values set up the encoding so that core knows what feature was being used and the version information

Reply:
What do you mean? how can I send different features if that can't be overridden?

Reply:
These are constants-
sdkSemver: APP_VERSION,
techVersion: VERSION.full,
sdkCode: 'K',

They are not supposed to be overridden. For example, if we send the above to URL in core along with accessiblity true then we know we need to append the feature as well. That information is taken from core.. not angular

Reply:
Then in core we pass in analyticsOptions to the method getAnalyticsOptions(options);
Take a look at https://github.com/cloudinary/cloudinary_js/pull/227/files#diff-5ebe1ce7b7707a2764175889bb5e4643R6

Reply:
Well, I get it, but still don't understand why this `defaultOptions` structure is found so many times in the code. it should be a single const, used once globally and sent with every call to core

Reply:
It is. Take a look here projects/angular-cld/src/lib/analyticsOptionsDefault.ts. 
We only pass it to core when we call core to build a url